modules/seqexec/server/src/main/scala/seqexec/server/gpi/GPIHeader.scala

```diff
@@ -4,26 +4,26 @@
 package seqexec.server.gpi
 
 import gem.Observation
+import cats.effect.IO
 import seqexec.model.dhs.ImageFileId
-import seqexec.server.SeqAction
+import seqexec.server.{InstrumentSystem, SeqAction}
 import seqexec.server.keywords._
 import seqexec.server.tcs.TcsKeywordsReader
 import seqexec.server.tcs.CRFollow
 
 object GPIHeader {
 
-  def header[A: HeaderProvider](inst: A,
-                                gdsClient: GDSClient,
-                                tcsKeywordsReader: TcsKeywordsReader,
-                                obsKeywordsReader: ObsKeywordsReader): Header =
+  def header(inst: InstrumentSystem[IO],
+             gdsClient: GDSClient,
+             tcsKeywordsReader: TcsKeywordsReader,
+             obsKeywordsReader: ObsKeywordsReader): Header =
     new Header {
       override def sendBefore(obsId: Observation.Id,
                               id: ImageFileId): SeqAction[Unit] = {
-        val ks = bundleKeywords(
-          inst,
+        val ks = inst.keywordsClient.bundleKeywords(
           List(
             buildDouble(tcsKeywordsReader.getParallacticAngle
-                          .map(_.map(_.toDoubleDegrees))
+                          .map(_.map(_.toSignedDoubleDegrees))
```

Comment:
This returns a double with many digits of precision, like -22.291460009476502. We are going to leave it this way as the same amount of digits is used for other values.

This is up for review though

Reply:
What is your concern re: very precise angles? Are you concerned that the TCS won't be able to parse them or something?

Reply:
it is a "change" in the sense that FITS files prevously only had 3 decimals. I'm not sure if there is any effect downstream